### ISSUE TITLE
word embeddings with DrQA

### ISSUE THREAD

**Issue by bittlingmayer** [facebookresearch/ParlAI#100]
From the below, I cannot really tell what is happening with the word embeddings by default.

From the warning, it seems like it should train on the text given (similar to fasttext unsupervised).

But from the param value pretrained_words True, it seems like it is using some pretrained .vec that was never provided.

Any clarification?

```
05/25/2017 02:17:53 PM: Setting fix_embeddings to False as embeddings are random.
05/25/2017 02:17:53 PM: [ Initializing model from scratch ]
05/25/2017 02:17:56 PM: [ WARNING: No embeddings provided. Keeping random initialization. ]
05/25/2017 02:17:56 PM: [ Created with options: ] 
no_images	False
num_features	3
question_merge	self_attn
pretrained_words	True
rnn_type	lstm
dict_max_ngram_size	-1
vocab_size	146272
doc_layers	3
use_in_question	True
embedding_dim	300

```

**Comment by ajfisch** [comment]
I'm sorry for the confusion, the code could use some updates.

There are a couple different options here that overlap.

`fix_embeddings` will remove the gradient to the embedding layer completely.
`tune_partial` will remove the gradient to all embeddings but the N most frequent words.
`embedding_file` specifies a pre-trained .txt of `word<space>dim0<space>...<space>dimN` embeddings.
`pretrained_words` says to only use words in the embedding_file, all other words are UNK. If false, there are a mix of random and pre-trained embeddings.

If there is no embedding_file, pretrained_words is ignored and embeddings are learned.

**Comment by bittlingmayer** [comment]
Thanks

To be clear: UNK is assigned a different random embedding each time.

**Comment by ajfisch** [comment]
Correct.
